packer/HWIMO-BUILD

```diff
@@ -13,18 +19,71 @@
 # early exit on command failure
 set -e
 
-#Ubuntu Version, can become an ENV VAR in the future
+echo '[Info] Please use ENV VAR to pass customized parameter to this script. Please refer to the usage on the top of script file.'
+
+############################  Parameter Process ################ start #############
+
+#Ubuntu Version
 if  [ ! -n "${OS_VER}" ];  then
     OS_VER=ubuntu-14.04
 fi
-echo "Packer Build based on $OS_VER"
+echo "[Info] Packer Build based on $OS_VER"
 
-# Build Vagrant or VMware OVA
+# Build Vagrant or VMware OVA ?
 if  [ ! -n "${BUILD_TYPE}" ];  then
-    BUILD_TYPE=vmware  # option is vmware or vitrualbox , case senstive
+    BUILD_TYPE=virtualbox  # option is vmware or vitrualbox , case senstive
+fi
+
+echo "[Info] Packer Build : ${BUILD_TYPE}"
+
+
+IS_UPLOAD_ATLAS=false
+if [[ "${UPLOAD_BOX_TO_ATLAS}" == "true" ]] ||  [[ "${UPLOAD_BOX_TO_ATLAS}" == "TRUE" ]];
+then
+    IS_UPLOAD_ATLAS=true
+fi
+if [ $BUILD_TYPE != "virtualbox" ] && [ $IS_UPLOAD_ATLAS == true ]; then
+    echo "[Error]: Bad Env Var: only when for Vagrant Box build. ( BUILD_TYPE=virtualbox), the $UPLOAD_BOX_TO_ATLAS can be set to true . Abort !"
+    exit 1
 fi
+if [ $IS_UPLOAD_ATLAS == true ]; then
+   if [ ! -n "${ATLAS_USERNAME}" ] || [ ! -n "${ATLAS_NAME}" ] || [ ! -n "${ATLAS_TOKEN}" ];  then
+       echo "[Error]: Bad Env Var: When ENV VAR \$UPLOAD_BOX_TO_ATLAS is true, the ATLAS_USERNAME/ATLAS_NAME/ATLAS_TOKEN should be given.  Abort !"
+       exit 2
+   fi
+   if [ ! -n "${ATLAS_VERSION}" ];  then
+       ATLAS_VERSION=""
+   else
+       if [[ $ATLAS_VERSION != +([0-9]).+([0-9]).+([0-9]) ]] ; then
+           echo "[Error]: ATLAS_VERSION(${ATLAS_VERSION}) should be X.Y.Z (X.Y.Z is digit), that's only format accepted by ATLAS."
+          exit 2
+      fi
+   fi
+fi
+
+
+# By default. use ansible playbook : rackhd_local.yml
+# can become an ENV VAR in the future
+if  [ ! -n "${RACKHD_INSTALL_ANSIBLE}" ];  then
```

Comment:
Is it necessary to check "RACKHD_INSTALL_ANSIBLE" in [rackhd_ci_builds,  rackhd_release,  rackhd_local] ?

Or check whether the script $RACKHD_INSTALL_ANSIBLE.yml exists?